cirq/contrib/qasm_import/_lexer_test.py

```diff
@@ -59,6 +61,36 @@ def test_valid_ids(identifier: str):
     assert token.value == identifier
 
 
+@pytest.mark.parametrize('number', [
+    '1e2',
+    '1e0',
+    '3.',
+    '4.e10',
+    '.333',
+    '1.0',
+    '0.1',
+    '2.0e-05',
+    '1.2E+05',
+    '123123.2132312',
+])
+def test_numbers(number: str):
+    lexer = QasmLexer()
+    lexer.input(number)
+    token = lexer.token()
+
+    assert token is not None
+    assert token.type == "NUMBER"
+    assert token.value == Number(number)
+
+
+def test_pi():
+    lexer = QasmLexer()
+    lexer.input('pi')
+    token = lexer.token()
+    assert token.type == "PI"
+    assert token.value == sympy.pi
```

Comment:
Why is this being parsed as a symbolic constant instead of just a float?

Reply:
Originally I parsed parameters with symbolic expressions so that circuits can be more human readable when printed out. However, due to #1611 I had to convert them back to float at the very end (when applying parameters for the gate). I we see no chance for symbolic mode for circuits then I'm happy to switch over to floats.